From 35db4a2433fbdf8612cf98b5aab5b14aeb5372f8 Mon Sep 17 00:00:00 2001
From: Tim Philipp <tim@centricular.com>
Date: Fri, 18 Oct 2019 00:39:12 +0100
Subject: [PATCH] meson: build gir even when cross-compiling if introspection
 was enabled explicitly

This can be made to work in certain circumstances when
cross-compiling, so default to not building g-i stuff
when cross-compiling, but allow it if introspection was
enabled explicitly via -Dintrospection=enabled.

See gstreamer/gstreamer#454 and gstreamer/gstreamer#381.

Upstream commit: 97a4d0cd7de8cfbf983acc7e37ba2f8fb73c3e19
Signed-off-by: Tim Philipp <tim@centricular.com>
Signed-off-by: Adam Duskett <aduskett@gmail.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 67aeeb4d03..7b3314bd9f 100644
--- a/meson.build
+++ b/meson.build
@@ -405,7 +405,7 @@ python3 = import('python').find_installation()
 
 gir = find_program('g-ir-scanner', required : get_option('introspection'))
 gnome = import('gnome')
-build_gir = gir.found() and not meson.is_cross_build()
+build_gir = gir.found() and (not meson.is_cross_build() or get_option('introspection').enabled())
 gir_init_section = [ '--add-init-section=extern void gst_init(gint*,gchar**);' + \
     'g_setenv("GST_REGISTRY_1.0", "@0@", TRUE);'.format(meson.current_build_dir() + '/gir_empty_registry.reg') + \
     'g_setenv("GST_PLUGIN_PATH_1_0", "", TRUE);' + \
-- 
2.24.1

